server/src/Types.hs

```diff
@@ -153,16 +156,37 @@ newtype AppliedScript = AppliedScript Ledger.Script
   deriving stock (Show, Generic)
   deriving newtype (Eq, FromJSON, ToJSON)
 
+data HashBytesRequest = HashBytesRequest 
+  { methodToUse :: HashMethod 
+  , bytesToHash :: BytesToHash
+  }
+  deriving stock (Show, Eq, Generic)
+  deriving anyclass (FromJSON, ToJSON)
+
 newtype BytesToHash = BytesToHash ByteString
   deriving stock (Show, Generic)
   deriving newtype (Eq)
   deriving (FromJSON, ToJSON) via JsonHexString
 
-newtype Blake2bHash = Blake2bHash ByteString
+newtype HashedBytes = HashedBytes ByteString
   deriving stock (Show, Generic)
   deriving newtype (Eq)
   deriving (FromJSON, ToJSON) via JsonHexString
 
+data HashMethod = Blake2b_256
+                | Sha2_256
+                | Sha3_256
+  deriving stock (Show, Eq, Generic)
+  deriving anyclass (FromJSON, ToJSON)
```

Comment:
I think using something like: 
```haskell
instance FromJSON HashMethod where
  parseJSON = 
    genericParseJSON 
      defaultOptions 
      { constructorTagModifier = fmap toLower }
```
would make it nicer to use the endpoint (so that users can write the hash method in a more natural way)